### PR TITLE
app/vmalert: add series fetched for alerting rule

### PR DESCRIPTION
**Pull request by Haleygo**
part of https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4039

get seriesFetched data when query alerting rule if datasource support and expose `vmalert_alerting_rules_last_evaluation_series_fetched`
image
image

### CODE REVIEW

**Comment by valyala** [comment]
@Haleygo , thanks for the pull request!

@dmitryk-dk , @Amper , @f41gh7 , could you help reviewing this pull request?

File: app/vmalert/datasource/vm_prom_api.go [hunk 1]
```diff
@@ -22,6 +20,9 @@ type promResponse struct {
 		ResultType string          `json:"resultType"`
 		Result     json.RawMessage `json:"result"`
 	} `json:"data"`
+	Stats struct {
+		SeriesFetched int64 `json:"seriesFetched,omitempty"`
+	} `json:"stats,omitempty"`
```
Comment (dmitryk-dk) [inline app/vmalert/datasource/vm_prom_api.go:25]: i think it is better to create separate struct for better code readability

File: app/vmalert/datasource/vm_test.go [hunk 2]
```diff
@@ -78,6 +78,8 @@ func TestVMInstantQuery(t *testing.T) {
 			w.Write([]byte(`{"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"vm_rows","foo":"bar"},"value":[1583786142,"13763"]},{"metric":{"__name__":"vm_requests","foo":"baz"},"value":[1583786140,"2000"]}]}}`))
 		case 7:
 			w.Write([]byte(`{"status":"success","data":{"resultType":"scalar","result":[1583786142, "1"]}}`))
+		case 8:
+			w.Write([]byte(`{"status":"success","data":{"resultType":"scalar","result":[1583786142, "1"]},"stats":{"seriesFetched":2}}`))
```
Comment (dmitryk-dk) [inline app/vmalert/datasource/vm_test.go:82]: As I can see, there is only one series fetched. Please change the field value to be consistent

**Review by dmitryk-dk — COMMENTED** [review]
Overall looks good. Please check some comments.

File: app/vmalert/datasource/vm_prom_api.go (hunk 1 above)
Comment (Haleygo) [reply app/vmalert/datasource/vm_prom_api.go:25]: Got it, but I see other fileds like `promResponse.Data` and `promInstant.Result` around `Stats` also not separate struct, should I separate them all?

Comment (Haleygo) [reply app/vmalert/datasource/vm_prom_api.go]: all fixed

File: app/vmalert/datasource/vm_test.go (hunk 2 above)
Comment (Haleygo) [reply app/vmalert/datasource/vm_test.go]: right! fixed

**Comment by Haleygo** [comment]
> Overall looks good. Please check some comments.

@dmitryk-dk Rephrased some comments, can you help reviewing?
Let me know if there is any misunderstanding

**Comment by Haleygo** [comment]
@f41gh7 can you help reviewing this?

**Comment by hagen1778** [comment]
Hello @Haleygo! Thanks for the initial implementation! 
I made some updates in the following PR https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4198
Please, take a look!

**Pull request closed without merging**